lib/cosmos/dart/lib/dart_database_cleaner.rb

```diff
@@ -31,9 +31,52 @@ def self.clean(force, full = false)
     Cosmos::Logger::info("Database cleanup complete!")
   end
 
-  def self.remove_packet_log(filename)
+  def remove_packet_log(filename)
     filename = filename.gsub("\\", "/") # Fix slashes
     filename = File.expand_path(filename, Cosmos::System.paths['DART_DATA']) # Make absolute path
+    if File.exists?(filename)
+      packet_log = PacketLog.where("filename = ?", filename).first
+      if packet_log
+        size = File.size(filename)
+        reader = Cosmos::PacketLogReader.new
+        reader.open(filename)
+        first_packet = nil
+        last_packet = nil
+        start_time = nil
+        end_time = nil
+        begin
+          first_packet = reader.first
+          last_packet = reader.last
+          start_time = first_packet.received_time
+          end_time = last_packet.received_time
+        rescue
+          if size == 128 or size == 0
+            Cosmos::Logger::error("File contains no packets: #{filename}")
+          else
+            Cosmos::Logger::error("Error analyzing file: #{filename}")
+          end
+          exit(1)
+        ensure
+          reader.close
+        end
+        num_deleted = PacketLogEntry.where("time >= ? and time <= ? and packet_log_id = ?", start_time - 1.minute, end_time + 1.minute, packet_log.id).delete_all
```

Comment:
What's with the +- 1 minute

Reply:
This just provides some tolerance incase the timestamps in the log were a little out of order. The packet_log_id constraint makes sure nothing extra is deleted.